File: 04_nelinearni_regresni_model_m-h_algoritmus/support/lik_ces.py
# Verohodnostni funkcep pro CES produkcni funkci

import numpy as np
from numpy import transpose as t
import math

def lik_ces(
    y: np.ndarray,
    x: np.ndarray,
    gamma: np.ndarray,
    h: float):

    f_x = gamma[0][0] * (gamma[1][0] * x[:,[1]]**gamma[3][0] + gamma[2][0] * x[:,[2]]**gamma[3][0])**(1 / gamma[3][0])
    n = y.shape[0]
    log_res = -n/2 * math.log(2 * math.pi) + n/2 * math.log(h) - h/2 * t(y - f_x) @ (y - f_x)
    res = math.exp(log_res)
    
    return res

File: Témata/04 nelineární regresní model (M-H algoritmus)/Matlab/lik_CES.m
function res = lik_CES(y,X,gamma,h)
%Verohodnostni funkce pro CES produkcni funkci
fx = gamma(1)*(gamma(2)*X(:,2).^gamma(4)+gamma(3)*X(:,3).^gamma(4)).^(1/gamma(4));
N = length(y);
log_res = -N/2*log(2*pi)+N/2*log(h)-h/2*(y-fx)'*(y-fx);
res = exp(log_res);
end

